projects/opendr_ws_2/src/opendr_perception/opendr_perception/skeleton_based_action_recognition_node.py

```diff
@@ -0,0 +1,243 @@
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-i", "--input_rgb_image_topic", help="Topic name for input image",
+                        type=str, default="/usb_cam/image_raw")
+    parser.add_argument("-o", "--output_rgb_image_topic", help="Topic name for output annotated image",
+                        type=lambda value: value if value.lower() != "none" else None,
+                        default="/opendr/image_pose_annotated")
+    parser.add_argument("-p", "--pose_annotations_topic", help="Topic name for pose annotations",
+                        type=lambda value: value if value.lower() != "none" else None,
+                        default="/opendr/poses")
+    parser.add_argument("-c", "--output_category_topic", help="Topic name for recognized action category",
+                        type=lambda value: value if value.lower() != "none" else None,
+                        default="/opendr/skeleton_recognized_action")
+    parser.add_argument("-d", "--output_category_description_topic", help="Topic name for description of the "
+                                                                          "recognized action category",
+                        type=lambda value: value if value.lower() != "none" else None,
+                        default="/opendr/skeleton_recognized_action_description")
+    parser.add_argument("--device", help="Device to use, either \"cpu\" or \"cuda\"",
+                        type=str, default="cuda", choices=["cuda", "cpu"])
+    parser.add_argument("--model", help="Model to use, either \"stgcn\" or \"pstgcn\"",
+                        type=str, default="stgcn", choices=["stgcn", "pstgcn"])
+
+    args = parser.parse_args()
+
+    try:
+        if args.device == "cuda" and torch.cuda.is_available():
+            device = "cuda"
+        elif args.device == "cuda":
+            print("GPU not found. Using CPU instead.")
+            device = "cpu"
+        else:
+            print("Using CPU.")
+            device = "cpu"
+    except:
+        print("Using CPU.")
+        device = "cpu"
+
+    skeleton_action_recognition_node = \
+        SkeletonActionRecognitionNode(input_rgb_image_topic=args.input_rgb_image_topic,
+                                      output_rgb_image_topic=args.output_rgb_image_topic,
+                                      pose_annotations_topic=args.pose_annotations_topic,
+                                      output_category_topic=args.output_category_topic,
+                                      output_category_description_topic=args.output_category_description_topic,
+                                      device=device,
+                                      model=args.model)
+
+    rclpy.spin(skeleton_action_recognition_node)
+
+    # Destroy the node explicitly
+    # (optional - otherwise it will be done automatically
+    # when the garbage collector destroys the node object)
+    skeleton_action_recognition_node.destroy_node()
+    rclpy.shutdown()
```

Comment:
In ROS2 we need to define a main function. Also fixed the default usb cam topic in the suggestion below.

```suggestion
def main(args=None):
    rclpy.init(args=args)

    parser = argparse.ArgumentParser()
    parser.add_argument("-i", "--input_rgb_image_topic", help="Topic name for input image",
                        type=str, default="image_raw")
    parser.add_argument("-o", "--output_rgb_image_topic", help="Topic name for output annotated image",
                        type=lambda value: value if value.lower() != "none" else None,
                        default="/opendr/image_pose_annotated")
    parser.add_argument("-p", "--pose_annotations_topic", help="Topic name for pose annotations",
                        type=lambda value: value if value.lower() != "none" else None,
                        default="/opendr/poses")
    parser.add_argument("-c", "--output_category_topic", help="Topic name for recognized action category",
                        type=lambda value: value if value.lower() != "none" else None,
                        default="/opendr/skeleton_recognized_action")
    parser.add_argument("-d", "--output_category_description_topic", help="Topic name for description of the "
                                                                          "recognized action category",
                        type=lambda value: value if value.lower() != "none" else None,
                        default="/opendr/skeleton_recognized_action_description")
    parser.add_argument("--device", help="Device to use, either \"cpu\" or \"cuda\"",
                        type=str, default="cuda", choices=["cuda", "cpu"])
    parser.add_argument("--model", help="Model to use, either \"stgcn\" or \"pstgcn\"",
                        type=str, default="stgcn", choices=["stgcn", "pstgcn"])

    args = parser.parse_args()

    try:
        if args.device == "cuda" and torch.cuda.is_available():
            device = "cuda"
        elif args.device == "cuda":
            print("GPU not found. Using CPU instead.")
            device = "cpu"
        else:
            print("Using CPU.")
            device = "cpu"
    except:
        print("Using CPU.")
        device = "cpu"

    skeleton_action_recognition_node = \
        SkeletonActionRecognitionNode(input_rgb_image_topic=args.input_rgb_image_topic,
                                      output_rgb_image_topic=args.output_rgb_image_topic,
                                      pose_annotations_topic=args.pose_annotations_topic,
                                      output_category_topic=args.output_category_topic,
                                      output_category_description_topic=args.output_category_description_topic,
                                      device=device,
                                      model=args.model)

    rclpy.spin(skeleton_action_recognition_node)

    # Destroy the node explicitly
    # (optional - otherwise it will be done automatically
    # when the garbage collector destroys the node object)
    skeleton_action_recognition_node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

```